src/Http/Client.php

```diff
@@ -161,7 +161,7 @@ public function __construct($config = [])
     {
         $this->setConfig($config);
 
-        $adapter = $this->_config['adapter'];
+        $adapter = $this->_defaultConfig['adapter'];
```

Comment:
How would someone change the adapter? I recognize that is not possible right now, but given the https proxy issues PHP streams have I plan to build a curl adapter in the near future.

Reply:
Yes, after I had submitted this pull request, I thought a little more. I had the same thought, that perhaps the intended functionality was to set the default to the config before merging the userland configuration.

Would it be worth updating to something like this?
```php
        $adapter = $this->_defaultConfig['adapter'];
        $this->setConfig('adapter', $adapter);
        $this->setConfig($config);

        //$adapter = $this->_defaultConfig['adapter'];
        $this->setConfig('adapter', null);
        if (is_string($adapter)) {
```